### PR TITLE
Connect api to forum

### PR DESCRIPTION
**Pull request by secondaryfun**
- Related Issue (include '#'):  https://github.com/CotripperPlatform/CoTrip/issues/684

- Description of changes made:

The following features of the forum-page-hashtag & forum-page-topic pages are now loading from the backend:
* Posts, heading, detail, followers

The search bar now functions and switches between topics/hashtags.

- Is the feature complete/bug resolved/etc..:  Still to be implemented:

* Media is not connected. Still loading from a placeholder.

- Any known bugs/strange behavior:  No

- Is there specific feedback you would like on these changes:  No

- Screenshot(s):

image
image

### CODE REVIEW

File: client/src/components/Modal/Modal.js [hunk 1]
```diff
@@ -2,7 +2,7 @@ import React from "react";
 import "./Modal.css";
 
 const Modal = props => {
-  console.log(props);
+  // console.log(props);
```
Comment (0xZakk) [inline client/src/components/Modal/Modal.js]: Delete commented out code

File: client/src/components/NavLink/NavLink.js [hunk 2]
```diff
@@ -19,7 +19,7 @@ class NavLink extends React.Component {
     }
   }
   render() {
-    console.log(this.props);
+    // console.log(this.props);
```
Comment (0xZakk) [inline client/src/components/NavLink/NavLink.js]: Delete

File: client/src/pages/ForumPage/ForumPage.js [hunk 3]
```diff
@@ -60,7 +60,9 @@ const ForumPage = props => {
           />
         </div>
         <div>
-          <header className="ForumPage__header">My Favorite Topics</header>
+          <header className="ForumPage__header">My Favorite Topics:
+          <a href="/forum-page-topic/1">Example Topic Detail</a>
```
Comment (0xZakk) [inline client/src/pages/ForumPage/ForumPage.js]: Can we remove this?

File: client/src/pages/ForumPage/ForumPage.js [hunk 4]
```diff
@@ -166,7 +168,7 @@ const ForumPage = props => {
         </div>
         <div>
           <header className="ForumPage__header">
-            #hashtags I'm Following: <a href="/forum-page-hashtag">Example Tag</a>
+            #hashtags I'm Following: <a href="/forum-page-hashtag/1">Example Tag</a>
```
Comment (0xZakk) [inline client/src/pages/ForumPage/ForumPage.js]: This too? I think @Bcastillo44  is working on a PR related to having the pills actually work

**Review by 0xZakk — CHANGES_REQUESTED** [review]

File: client/src/components/Modal/Modal.js (hunk 1 above)
Comment (secondaryfun) [reply client/src/components/Modal/Modal.js]: Done.

File: client/src/components/NavLink/NavLink.js (hunk 2 above)
Comment (secondaryfun) [reply client/src/components/NavLink/NavLink.js]: Done.

File: client/src/pages/ForumPage/ForumPage.js (hunk 3 above)
Comment (secondaryfun) [reply client/src/pages/ForumPage/ForumPage.js]: Done.

File: client/src/pages/ForumPage/ForumPage.js (hunk 4 above)
Comment (secondaryfun) [reply client/src/pages/ForumPage/ForumPage.js]: @Bcastillo44  will have a whole new page.  I was just updating the placeholder.

**Comment by secondaryfun** [comment]
I removed any reference to ForumPost since @Bcastillo44 is working on that.  To test this issue, go to 'forum-page-hashtag/1'

**Review by 0xZakk — CHANGES_REQUESTED** [review]
@Bcastillo44

**Pull request merged**